File: language-support/js/daml-ledger/src/mapping/submit_and_wait_request.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as mapping from '.';
import * as ledger from '..';
import * as grpc from 'daml-grpc';

export const SubmitAndWaitRequest: mapping.Mapping<grpc.SubmitAndWaitRequest, ledger.SubmitAndWaitRequest> = {
    toObject(request: grpc.SubmitAndWaitRequest): ledger.SubmitAndWaitRequest {
        return {
            commands: mapping.Commands.toObject(request.getCommands()!)
        };
    },
    toMessage(request: ledger.SubmitAndWaitRequest): grpc.SubmitAndWaitRequest {
        const result = new grpc.SubmitAndWaitRequest();
        result.setCommands(mapping.Commands.toMessage(request.commands));
        return result;
    }
}

File: canton/community/bindings-java/src/main/java/com/daml/ledger/javaapi/data/SubmitAndWaitRequest.java
// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates.
// Proprietary code. All rights reserved.

package com.daml.ledger.javaapi.data;

import com.daml.ledger.api.v2.CommandServiceOuterClass;
import org.checkerframework.checker.nullness.qual.NonNull;

public final class SubmitAndWaitRequest {

  public static CommandServiceOuterClass.SubmitAndWaitRequest toProto(
      @NonNull CommandsSubmission submission) {
    return CommandServiceOuterClass.SubmitAndWaitRequest.newBuilder()
        .setCommands(submission.toProto())
        .build();
  }

  public static CommandsSubmission fromProto(
      CommandServiceOuterClass.@NonNull SubmitAndWaitRequest request) {
    return CommandsSubmission.fromProto(request.getCommands());
  }
}
